### PR TITLE
clone map to make sample immutable

### PR DESCRIPTION
**Pull request by simonleung8**
Currently command is passed to sample as a pointer, so changing command will change the data within sample.

This change will make a copy of command before passing to sample, hence sample is now immutable

### CODE REVIEW

File: experiment/runner.go [hunk 1]
```diff
@@ -192,7 +200,7 @@ func (ex *SamplableExperiment) Sample() {
 			workers = workers + w
 		case _ = <-heartbeat.C:
 			//heatbeat for updating CLI Walltime every second	
-		}
-		ex.samples <- &Sample{commands, avg, totalTime, iterations, totalErrors, workers, lastResult, lastError, worstResult, ninetyfifthPercentile, time.Now().Sub(startTime), sampleType}
+		}		
+		ex.samples <- &Sample{clone(commands), avg, totalTime, iterations, totalErrors, workers, lastResult, lastError, worstResult, ninetyfifthPercentile, time.Now().Sub(startTime), sampleType}
```
Comment (julz) [inline experiment/runner.go:204]: given that we had a bug, let's add a test that the Sample is immutable?

**Comment by simonleung8** [comment]
added tests for immutable sample history

**Comment by julz** [comment]
no need to test/export Clone, just leave it private (private methods don't need tests); apart from that looks good

**Comment by simonleung8** [comment]
updated code.  took out test for clone and function is now private

**Pull request merged**